.github/workflows/godot_cpp_test.yml

```diff
@@ -21,49 +21,42 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
         with:
-          submodules: recursive
-
-      - name: Setup Python and SCons
-        uses: ./.github/actions/godot-deps
+          sparse-checkout: .github
 
-      # Checkout godot-cpp
       - name: Checkout godot-cpp
         uses: actions/checkout@v4
         with:
+          submodules: recursive
           repository: godotengine/godot-cpp
           ref: ${{ env.GODOT_CPP_BRANCH }}
-          submodules: recursive
           path: godot-cpp
 
-      # Download generated API dump
+      - name: Setup Python and SCons
+        uses: ./.github/actions/godot-deps
+
+      - name: Setup GCC problem matcher
+        uses: ammaraskar/gcc-problem-matcher@master
+
       - name: Download GDExtension interface and API dump
         uses: ./.github/actions/download-artifact
         with:
           name: godot-api-dump
-          path: ./godot-api
+          path: ./godot-cpp/gdextension
 
-      # Extract and override existing files with generated files
-      - name: Extract GDExtension interface and API dump
-        run: |
-          cp -f godot-api/gdextension_interface.h godot-cpp/gdextension/
-          cp -f godot-api/extension_api.json godot-cpp/gdextension/
-
-      # TODO: Add caching to the SCons build and store it for CI via the godot-cache
-      # action.
+      - name: Restore Godot build cache
+        uses: ./.github/actions/godot-cache-restore
+        with:
+          cache-name: godot-cpp
+        continue-on-error: true
 
-      # Build godot-cpp test extension
       - name: Build godot-cpp test extension
-        run: |
-          cd godot-cpp/test
-          scons target=template_debug dev_build=yes
-          cd ../..
+        env: # Keep synced with godot-build.
+          SCONS_CACHE: ${{ github.workspace }}/.scons-cache/
+          SCONS_CACHE_LIMIT: 7168
```

Comment:
Doesn't really affect anything but the godot-cpp build system doesn't have any cache limiting, so we'd have to see how this can be handled safely, but we do use the cache in godot-cpp so it shouldn't matter but just need to make sure it doesn't tax our CI too much by filling up our limits

Reply:
Though one point is relevant, there's no configuration of the scons cache path in godot-cpp so unsure if this is guaranteed to be effective

Reply:
Note that just from this PR the cache upload size was 825 mb for a single run, that's a lot for the 10 gig cache limit, and might cause important other cache entries to be dropped and hurt general compile times

I think that, until we've implemented a cache limiting system in godot-cpp we should avoid using the cache here

Reply:
Hmm, fair point; haven't considered any cache limiting techniques used here. For now, I'll keep those sections commented out and add a `TODO` note.

Reply:
I'll take a look at adding cache pruning to godot-cpp over the next few days and we can see how it works out